src/domain/map/HankeMapComponent.tsx

```diff
@@ -53,13 +53,22 @@ const HankeMapComponent: React.FC<Props> = ({
 
   const [zoom] = useState(9); // TODO: also take zoom into consideration
 
+  const [hankeSources] = useState<VectorSource>(new VectorSource());
+
   const hankkeetFilteredByAll =
-    (!loadingProjects || loadingProjectsError) && projectsData && Array.isArray(projectsData.data)
-      ? projectsData.data.filter(
+    (!loadingProjects || loadingProjectsError) && projectsData && Array.isArray(projectsData)
+      ? projectsData.filter(
           byAllHankeFilters({ startDate: hankeFilterStartDate, endDate: hankeFilterEndDate })
         )
       : [];
 
+  useEffect(() => {
+    hankkeetFilteredByAll.forEach((hanke) => {
+      if (hanke.geometriat)
+        hankeSources.addFeatures(new GeoJSON().readFeatures(hanke.geometriat.featureCollection));
+    });
+  }, [hankkeetFilteredByAll]);
```

Comment:
Voisiko projectsDatan laittaa tähän dependencynä projectsData ja tehdä filtteröinnin useEffectin sisällä? Ei tarvitsisi tallentaa erilliseen muuttujaan.

Reply:
Jos näin tehdään ja otetaan useRef hankeSourcesille käyttöön niin milloin ei triggeröi renderiä dateRangeControlin aikarajausten muuttamisesta 🤔

Reply:
Lisäksi tuota muuttujaa käytetään templatessa tukemaan testejä laskettaessa aikarajauksen vaikutuksia 🤔 
Sanoisin siis että hyvä idea, mutta tähän kokonaisuuteen en lähtisi viemään muutosta.

Reply:
Eipä tietääkseni koska projectsData tulee tällä propsina ja sisältää kaikki hankkeet -- se ei siis muutu kun aikarajaus muuttuu eikä näin ollen triggeröisi efektiä. Lisäksi tuota hankkeetFilteredByAll käytetään testaukseen --- käytetään templatessa sellaisenaan kertomaan filtteröityjen hankkeiden määrä -- eli käytössä kahdessa paikkaa.

Tämän myötä sanoisin että jättäisin sen nyt tuollaiseksi